demos/mqtt/mqtt_demo_connection_sharing/subscription-manager/mqtt_subscription_manager.c

```diff
@@ -0,0 +1,288 @@
+/*
+ * Copyright (C) 2020 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#include <string.h>
+#include "mqtt_subscription_manager.h"
+
+#define MAX_SUBSCRIPTION_CALLBACK_RECORDS    5
+
+typedef struct SubscriptionManager_Record
+{
+    const char * pTopicFilter;
```

Comment:
Which module is responsible for topic filter memory? Will application be responsible for all topic filter buffers if it subscribes to different topics?

Reply:
Yes, the single-threaded application will be responsible to maintain the lifetime of the topic string memory that is provides to the subscription manager. (Ideally, in a multi-threaded use-case, the data should be copied into the record, but I am thinking of keeping it simple for the single-threaded demo)